packages/react/src/components/Authenticator/shared/FormField.tsx

```diff
@@ -0,0 +1,61 @@
+import { FormFieldOptions } from '@aws-amplify/ui';
+import {
+  PasswordField,
+  PhoneNumberField,
+  TextField,
+} from '../../../primitives';
+import { useAuthenticator } from '../hooks/useAuthenticator';
+
+export interface FormFieldProps {
+  name: string;
+  formFieldOptions: FormFieldOptions;
+}
+export function FormField({ name, formFieldOptions }: FormFieldProps) {
+  const { validationErrors } = useAuthenticator();
+  const { type } = formFieldOptions;
+
+  if (type === 'tel') {
+    return (
+      <PhoneNumberField
+        name={name}
+        label={formFieldOptions.label}
+        placeholder={formFieldOptions.placeholder}
+        defaultCountryCode={formFieldOptions.dialCode}
+        countryCodeName="country_code"
+        dialCodeList={formFieldOptions.dialCodeList}
+        autoComplete={formFieldOptions.autocomplete as string}
```

Comment:
Is the casting due to my suggestion that we use a union of `string` and `boolean` for the type of `autocomplete`?

Reply:
Talked offline on this. I tried the suggestion

```tsx
type Autocomplete<T = string | boolean> = T extends string ? string : T extends boolean ? boolean : never;

const autocompleteString: Autocomplete = 'string';
const autocompleteBoolean: Autocomplete = true;
```
This works when assigning a value to a variable, but this did not work when passing a value to a string field:

```tsx
let val: Autocomplete;

// ... 

<TextField
  autoComplete={val} // doesn't work
/>
```

this wouldn't work because `Autocomplete` is still considered `string | boolean`.

Reply:
Thinking that for the time being we can just go back to typing `autocomplete` as `string` to remove the need for casting. FWIW think the above example would work by just passing `string` as the generic value to `Autocomplete`, e.g. `let val: Autocomplete<string>` but this can be addressed later, don't want to hold up your PR

Reply:
Got it, but let's tackle this when we do our refactor for react-native.